lib/puppet/type/cloudwatch_alarm.rb

```diff
@@ -0,0 +1,91 @@
+Puppet::Type.newtype(:cloudwatch_alarm) do
+  @doc = 'type representing an AWS CloudWatch Alarm'
```

Comment:
I like using `desc` to be consistent with how the types are documented, otherwise I'm not sure of the difference.